pipeline/backend/kubernetes/utils_test.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDNSName(t *testing.T) {
+	name, err := dnsName("wp_01he8bebctabr3kgk0qj36d2me_0_services_0")
+	assert.NoError(t, err)
+	assert.Equal(t, "wp-01he8bebctabr3kgk0qj36d2me-0-services-0", name)
+
+	name, err = dnsName("wp-01he8bebctabr3kgk0qj36d2me-0\\services-0")
+	assert.NoError(t, err)
+	assert.Equal(t, "wp-01he8bebctabr3kgk0qj36d2me-0\\services-0", name)
```

Comment:
Do we want those `\\` to be in the dns name? (Not that I expect the uuids to contain such chars 😅 )

Reply:
no idea but the current regex do allow it ...

Reply:
https://github.com/woodpecker-ci/woodpecker/blob/a63135363b513ba4bb13f0a721f4b7998cf3f85d/pipeline/backend/kubernetes/utils.go#L30

was added at #1661

@NeilHanlon might have more insight?

Reply:
looking at the change, it appears, though I was the last to touch that line, I didn't write the regex. If it's incorrect it was incorrect before I touched it... promise! :P

Reply:
Haha no problem thanks for the fast headsup :)

It would be interesting why exactly this regex and not something else.

I guess i can just simplify it :)

Reply:
in fact, i think that this is intended not to match `\\` but is just escaping the `\`, so it should be matching `\` followed by anything.. it's the 'followed by anything' bit that is problematic--and that should probably be `\\[a-z0-9]` without the `.`

Reply:
done